common/resource/resourceImpl.go

```diff
@@ -151,6 +152,11 @@ func New(
 		return nil, err
 	}
 
+	hostInfo, err := membershipMonitor.WhoAmI()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
does this works before starting `membershipMonitor`?

Reply:
Moved to resource.Start().